src/sink/rest.rs

```diff
@@ -518,18 +532,30 @@ async fn codec_task(
         host: hostname(),
         ..EventOriginUri::default()
     };
-
+    // create token in this lifetime (ambiguous to compiler), instead of bound to the sink.
+    let token = match auth {
+        Some(Auth::GCP) => {
+            let (t, _) = auth::authenticate_bearer().await?;
```

Comment:
I would completely avoid `authentication bearer here as there are other things wrong with it too (for example it's not async yet has an async definition) and you incure extra wait by making a blocking requestion, that you don't need to here.
```suggestion
            let t = Token::new()?;
```